internal/commands/push/command_test.go

```diff
@@ -1125,34 +1132,136 @@ func TestPushHandlerCreateNewApp(t *testing.T) {
+				procedure: func(c *expect.Console) {
+					c.ExpectString("Determining changes")
+					c.ExpectString("Deployed app is identical to proposed version, nothing to do")
+				},
+			},
+			{
+				description: "with a diff which is accepted by the user",
+				diffFn:      func(groupId, appID string, appData interface{}) ([]string, error) { return []string{"diff"}, nil },
+				procedure: func(c *expect.Console) {
+					c.ExpectString("Please confirm the changes shown above")
+					c.SendLine("y")
+
+					c.ExpectString("Determining changes")
+					c.ExpectString("Creating draft")
+					c.ExpectString("Deploying draft")
+					c.ExpectString("Deployment complete")
+					c.ExpectString("Successfully pushed app up: eggcorn-abcde")
+				},
+			},
+			{
+				description: "with a diff which is rejected by the user",
+				diffFn:      func(groupId, appID string, appData interface{}) ([]string, error) { return []string{"diff"}, nil },
+				procedure: func(c *expect.Console) {
+					c.ExpectString("Please confirm the changes")
+					c.SendLine("")
+				},
+			},
+		} {
+			t.Run(tc.description, func(t *testing.T) {
+				realmClient.DiffFn = tc.diffFn
+				tmpDir, teardown, tmpDirErr := u.NewTempDir("push_handler")
+				assert.Nil(t, tmpDirErr)
+				defer teardown()
+
+				app := local.App{RootDir: tmpDir, Config: local.FileRealmConfig, AppData: &local.AppRealmConfigJSON{local.AppDataV2{local.AppStructureV2{
+					ConfigVersion:   realm.AppConfigVersion20210101,
+					Name:            "eggcorn",
+					Location:        realm.Location("location"),
+					DeploymentModel: realm.DeploymentModel("deployment_model"),
+					Environment:     realm.Environment("environment"),
+				}}}}
+
+				app.WriteConfig()
+
+				_, console, _, ui, consoleErr := mock.NewVT10XConsole()
+				assert.Nil(t, consoleErr)
+				defer console.Close()
+
+				doneCh := make(chan (struct{}))
+				go func() {
+					defer close(doneCh)
+					console.ExpectString("Do you wish to create a new app?")
+					console.SendLine("y")
+
+					console.ExpectString("App Name")
+					console.SendLine("testApp")
+
+					console.ExpectString("App Location")
+					console.Send(string(terminal.KeyArrowDown))
+					console.SendLine("")
+
+					console.ExpectString("App Deployment Model")
+					console.Send(string(terminal.KeyArrowDown))
+					console.SendLine("")
+
+					console.ExpectString("App Environment")
+					console.Send(string(terminal.KeyArrowDown))
+					console.SendLine("")
+
+					console.ExpectString("Please confirm the new app details shown above")
+					console.SendLine("y")
+
+					console.ExpectString("App created successfully")
+
+					tc.procedure(console)
+
+					console.ExpectEOF()
+				}()
+
+				cmd := &Command{inputs{LocalPath: tmpDir, RemoteApp: "appID"}}
+				err := cmd.Handler(nil, ui, cli.Clients{Realm: realmClient})
+
+				console.Tty().Close() // flush the writers
+				<-doneCh              // wait for procedure to complete
+
+				assert.Nil(t, err)
```

Comment:
likewise here, there are very few times (if ever) in `go` that shouldn't be immediately doing something with the `err` returned from a func, so this can be moved up (or inlined) with the `cmd.Handler` call itself